### PR TITLE
Replace most osSyncPrintf calls with PRINTF macro

### PR DESCRIPTION
**Pull request by cadmic**
I replaced them everywhere in `src`, except:

* src/audio/general.c
* src/boot/assert.c
* src/boot/is_debug.c
* src/code/__osMalloc.c
* src/code/fault.c
* src/code/fault_drawer.c

The `audio` and `code` files still call osSyncPrintf in retail, and I feel `assert.c` and `is_debug.c` didn't really make sense to change but I'm not sure.

Thanks to @Roman971 for the idea to fake variadic macros by suppressing IDO's warning.

### CODE REVIEW

**Review by fig02 — APPROVED** [review]

**Review by Thar0 — APPROVED** [review]

**Comment by fig02** [comment]
"I feel assert.c and is_debug.c didn't really make sense to change but I'm not sure"
Do we know if these print calls are removed from retail? If they are, I think it still makes sense to use the macro here

**Comment by cadmic** [comment]
In assert.c, `osSyncPrintf` is called from `__assert` which is removed entirely in retail. Similarly, in is_debug.c, `osSyncPrintf` is called in `func_80002384` which is removed entirely in retail. But both of these uses seem very "low-level" debugging functions so I think it's reasonable if they call osSyncPrintf directly, but I don't have a strong opinion about it

**Comment by Dragorn421** [comment]
Here's my view on debug macros:

When starting to write a program, you are given functions and documentation about them, like
`osSyncPrintf(string, ...args)`: print a message
`Matrix_NewMtx()`: allocs a new matrix for the frame

While developing the program, you would probably make a debug build, which adds goodies to using some functions. (the details not mattering to the programmer)
For example `Matrix_NewMtx` checks for the coefficients to not be too big
Again the details of the debug checks or even the debug checks happening at all are details the programmer doesn't care about

When the program works well, you take off the training wheels and make a release build, without debug features. If needs be to debug something reproducible you can build a debug build again.

What I'm getting at is that from the programmer's standpoint there shouldn't be both a debug version of a function, and a release version. The programmer should have a single version to deal with, that to them looks like "release", the debug goodies should be added by cpp magic on debug builds.

With making macros that have a different name from the underlying implementation function, the programmer
- is given explicit knowledge they're using a debug wrapper, whereas you would typically make cpp magic transparent so existing code using release-like calls wouldn't need to change to the wrappers (e.g. imagine making `Matrix_RotateX` do something special in a debug build: the simplest is to make a `Matrix_RotateX` macro, not replace every `Matrix_RotateX`)
- could mistakenly use the unwrapped function directly (hence shortcutting the cpp magic)
- and documentation on the function would be confusing since there'd essentially be two names for it. The main one attached to the comments would be the implementation, yet the programmer is expected to use the differently-named debug wrapper


Thanks for reading my essay

**Comment by cadmic** [comment]
Personally, I would *like* to know that what I'm reading is actually a wrapper macro, which calls a different version of a function whether in debug or release mode. This is especially true for this project, since I'm interested in the exact differences between the debug and release builds. Maybe I would feel differently if we were starting our own project from scratch, but we don't get to do that here.

Here's an interesting example: in `src/audio/general.c`, two functions have `osSyncPrintf` left in the release build, while one function has the `osSyncPrintf` removed. It seems like the programmers themselves mistakenly used the unwrapped function directly. This is easily represented here by using `osSyncPrintf` for some calls and `PRINTF` for others. But if we call the macro `osSyncPrintf`, it could get awkward (do we `#undef` it and then `#define` it later with a copied definition?).

It's true that modders could mistakenly use the wrong function, but that wouldn't really be that bad, and tbh it doesn't feel worse than any other programming mistake that one could make. Besides I think people mostly would learn by example and pick up on things like `PRINTF` pretty quickly, similar to other macros like `OPEN_DISPS`/`CLOSE_DISPS`.

File: src/code/z_kanfont.c [hunk 1]
```diff
@@ -44,20 +44,20 @@ void Font_LoadOrderedFont(Font* font) {
 
     DmaMgr_RequestSyncDebug(font->msgBuf, (uintptr_t)_nes_message_data_staticSegmentRomStart + font->msgOffset, len,
                             "../z_kanfont.c", 122);
-    osSyncPrintf("msg_data=%x,  msg_data0=%x   jj=%x\n", font->msgOffset, font->msgLength, jj = len);
+    PRINTF("msg_data=%x,  msg_data0=%x   jj=%x\n", font->msgOffset, font->msgLength, jj = len);
 
     len = jj;
```
[the remark this answers is not in the document]
Comment (cadmic) [reply src/code/z_kanfont.c:49]: yeah, unfortunately I don't have a solution to this right now. Could we wait until we try to match the non-debug version of this file? It is suspicious that this printf call has too many arguments, which makes me think it is a fake match.

[the remark this answers is not in the document]
Comment (Dragorn421) [reply src/code/z_kanfont.c:49]: > this printf call has too many arguments
it doesn't? what am I missing

[the remark this answers is not in the document]
Comment (cadmic) [reply src/code/z_kanfont.c:49]: oh sorry, I was confused. I still think this should a problem for later though

**Comment by fig02** [comment]
In general I dont really agree with this notion that debug features should be "hidden from the programmer" and that the details of the debug features "dont matter" to the programmer. As programmers we care immensely about those features and want to use them to debug our problems.
Id like to know when looking at something that it is a macro and contains debug information that would be helpful for debugging.

My reasoning here is also why I think the macros should be in all caps instead of pretending to be "real" functions in https://github.com/zeldaret/oot/pull/1597

**Comment by fig02** [comment]
Jenkins failed on the latest commit, but it seems to be a jenkins agent issue. Hopefully will be fine on the next commit after merge conflicts are resolved

**Comment by cadmic** [comment]
Thanks for looking at that, I merged and jenkins is passing now

**Review by Dragorn421 — APPROVED** [review]

**Comment by fig02** [comment]
Merging tomorrow -- last call for reviews
(also theres more conflicts to resolve, sorry)

**Pull request merged**